Math/NumberTheory/GaussianIntegers.hs

```diff
@@ -181,50 +189,61 @@ a .^ e
     | otherwise = a * a .^ (e - 1)
     where
     s = a .^ div e 2
+{-# DEPRECATED (.^) "Use (^) instead." #-}
 
 -- |Compute the prime factorisation of a Gaussian integer. This is unique up to units (+/- 1, +/- i).
 -- Unit factors are not included in the result.
 factorise :: GaussianInteger -> [(GaussianInteger, Int)]
-factorise g = helper (Factorisation.factorise $ norm g) g
+factorise g = concat $ snd $ mapAccumL go g (Factorisation.factorise $ norm g)
```

Comment:
Is there a source for this algorithm (that might be worth citing from the doc comment)?

Reply:
The idea is to avoid repetitive complex divisions with the same divisor, because they unnecessary recalculate its norm. I'll refactor code to make it more palatable.

Reply:
Ah, I see. I think I was thrown off by the extra units that entered into the equations (to make the math work out nicer).